bundles/org.openhab.binding.luxtronikheatpump/src/main/java/org/openhab/binding/luxtronikheatpump/internal/HeatpumpConnector.java

```diff
@@ -0,0 +1,195 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.luxtronikheatpump.internal;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.Socket;
+import java.net.UnknownHostException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link HeatpumpConnector} reads / writes internal states of a Heat pump with Luxtronik control.
+ *
+ * Based on HeatpumpConnector class of novelanheatpump binding
+ * 
+ * @author Stefan Giehl - Initial contribution
+ */
+@NonNullByDefault
+public class HeatpumpConnector {
+
+    private static final int SOCKET_PARAM_WRITE_PARAMS = 3002;
+    private static final int SOCKET_PARAM_READ_PARAMS = 3003;
+    private static final int SOCKET_PARAM_READ_VALUES = 3004;
+    private static final int SOCKET_PARAM_READ_VISIBILITIES = 3005;
+
+    static final Logger logger = LoggerFactory.getLogger(HeatpumpConnector.class);
+
+    private String serverIp;
+    private int serverPort;
+    private Integer[] heatpumpValues = new Integer[0];
+    private Integer[] heatpumpParams = new Integer[0];
+    private Integer[] heatpumpVisibilities = new Integer[0];
+
+    public HeatpumpConnector(String serverIp, int serverPort) {
+        this.serverIp = serverIp;
+        this.serverPort = serverPort;
+    }
+
+    /**
+     * reads all values from the heatpump via network
+     * 
+     * @throws UnknownHostException indicate that the IP address of a host could not be determined.
+     * @throws IOException indicate that no data can be read from the heat pump
+     */
+    public void read() throws UnknownHostException, IOException {
+        try (Socket sock = new Socket(serverIp, serverPort)) {
```

Comment:
I'm not sure whether the protocol allows it to keep the connection alive. If yes, you could keep the connection open and reduce the risk of later issues when DNS or other network things are failing but the communication itself would work.

Reply:
I'm not able to answer that. The old binding and the other tools, I found for managing a luxtronik heatpump, are closing the connection again. In order to avoid any possible problems if the socket does not support that properly I wouldn't change that for now.